### PR TITLE
determine the verbatim pod indent from least indented line

### LINKED ISSUE #795: ASCII art broken in pod
#781 removes leading whitespace in pod verbatim sections. This breaks the ASCII art in pages like the [Growing Guide](http://mojolicio.us/perldoc/Mojolicious/Guides/Growing#Model-View-Controller). Interestingly it doesn't seem to break all of them, like the end of this [one](http://mojolicio.us/perldoc/Mojolicious/Guides/Growing#Differences). 

Not sure if we should revert the patch or harden the ASCII art. My initial reaction is that for a general purpose pod renderer we should leave leading whitespace as-is for purposes such as these.

### PR DESCRIPTION
**Pull request by jberger**
This addesses ascii art broken by #781

fixes #795

### CODE REVIEW

**Comment by jberger** [comment]
for the record, I would be as happy to unapply #781 as apply this patch

**Comment by jberger** [comment]
A variant that only computes `length` once, but it rather more ugly: http://pastie.org/10178896

File: lib/Mojolicious/Plugin/PODRenderer.pm [hunk 1]
```diff
@@ -96,7 +96,7 @@ sub _pod_to_html {
   my $parser = Pod::Simple::XHTML->new;
   $parser->perldoc_url_prefix('https://metacpan.org/pod/');
   $parser->$_('') for qw(html_header html_footer);
-  $parser->strip_verbatim_indent(sub { my ($i) = $_[0][0] =~ /^(\s+)/; $i });
+  $parser->strip_verbatim_indent(sub { (sort map /^(\s+)/, @{$_[0]})[0] });
```
Comment (kraih) [inline lib/Mojolicious/Plugin/PODRenderer.pm]: Is `map` without a block good style now?

Comment (jberger) [inline lib/Mojolicious/Plugin/PODRenderer.pm]: no, but without it perltidy kills it and makes it 4 lines

Comment (jberger) [inline lib/Mojolicious/Plugin/PODRenderer.pm]: that was actually my complaint here: http://irclog.perlgeek.de/mojo/2015-05-09#i_10574016

**Comment by jberger** [comment]
Note that @theory has actually adopted this logic in one of his modules. https://github.com/theory/text-markup/commit/6a8953d366389495ab519b2cc7bc142979ff8e08

**Pull request closed without merging**